src/lib.rs

```diff
@@ -105,51 +105,69 @@ pub mod sign;
 pub mod transaction;
 pub mod transport;
 
-use crate::contract::{
-    DeployBuilder, DeployedFuture, EventBuilder, EventStream, MethodBuilder, ViewMethodBuilder,
-};
-pub use crate::contract::{Event, Instance, Topic};
-pub use crate::int::I256;
-use crate::log::LogStream;
-pub use crate::transaction::{Account, GasPrice};
-pub use crate::transport::DynTransport;
+pub use crate::contract::Instance;
+pub use crate::prelude::*;
 pub use ethcontract_common as common;
 pub use ethcontract_common::truffle::Artifact;
 #[cfg(feature = "derive")]
 pub use ethcontract_derive::contract;
 pub use jsonrpc_core as jsonrpc;
-pub use secret::{Password, PrivateKey};
 pub use serde_json as json;
 pub use web3;
-use web3::api::Web3;
-pub use web3::types::{Address, BlockNumber, TransactionCondition, H160, H256, U256};
 
-/// Type alias for a `Web3` provider with an underlying `DynTransport`.
-pub type DynWeb3 = Web3<DynTransport>;
+pub mod prelude {
+    //! A prelude module for importing commonly used types when interacting with
+    //! generated contracts.
+
+    pub use crate::contract::{Event, Topic};
+    pub use crate::int::I256;
+    pub use crate::secret::{Password, PrivateKey};
+    pub use crate::transaction::{Account, GasPrice};
+    pub use web3::api::Web3;
+    pub use web3::transports::Http;
+    pub use web3::types::{Address, BlockNumber, TransactionCondition, H160, H256, U256};
+}
+
+pub mod dyns {
+    //! Type aliases to various runtime types that use an underlying
+    //! `DynTransport`. These types are used extensively throughout the
+    //! generated code.
+
+    use crate::contract::{
+        DeployBuilder, DeployedFuture, EventBuilder, EventStream, Instance, MethodBuilder,
+        ViewMethodBuilder,
+    };
+    use crate::log::LogStream;
+    pub use crate::transport::DynTransport;
+    use web3::api::Web3;
 
-/// Type alias for a contract `Instance` with an underyling `DynTransport`.
-pub type DynInstance = Instance<DynTransport>;
+    /// Type alias for a `Web3` with an underlying `DynTransport`.
+    pub type DynWeb3 = Web3<DynTransport>;
```

Comment:
I never really understood what the Dyn gives us over using Transport directly. Is this to allow extension methods?

Reply:
No, its for type-erasure. Its to make using the generated code more convenient by boxing the transport. So instead of having `MyContract<T>` everywhere, you just use `MyContract` which holds a boxed `T`.

Reply:
My question is more why we need the DynTransport in the first place instead of passing the generics along in our `expand_xyz` methods.

Reply:
It was originally just to make using the generated types a bit easier at the cost of some dynamic dispatch. In terms of code generation, it doesn't make much of a difference at all. As you said it would require just passing the `T: Transport` along.

I created #249 to make using dynamic dispatch opt-in.

Reply:
I have nothing against it, just wanted to understand its necessity.

Reply:
Absolutely not necessary, just to make using the generated contract types not require type parameters. Its one of those things I thought would be really cool but ended up not being super useful in the end.